pkg/service/ecloud/model.go

```diff
@@ -775,3 +787,21 @@ type HostSpec struct {
 	CPUClockSpeed int    `json:"cpu_clock_speed"`
 	RAMCapacity   int    `json:"ram_capacity"`
 }
+
+// Task represents a task against an eCloud resource
+// +genie:model_response
+// +genie:model_paginated
+type Task struct {
+	ID        string              `json:"id"`
+	Name      string              `json:"name"`
+	Status    TaskStatus          `json:"status"`
+	CreatedAt connection.DateTime `json:"created_at"`
+	UpdatedAt connection.DateTime `json:"updated_at"`
+}
+
+// TaskReference represents a reference to an on-going task
+// +genie:model_response
+type TaskReference struct {
+	TaskID     string `json:"task_id"`
+	ResourceID string `json:"id"`
```

Comment:
Should this be `ID` rather than `ResourceID`?

Reply:
I was in two minds, myself.  I thought this would help differentiate between the ID of the task and the ID of the resource when using this for different resource types.  Can easily update this if you prefer? :)

Reply:
Ahh cool, lets stick with `ResourceID` :)